### PR TITLE
Pension unit test coverage updates - QA

### PR DESCRIPTION
**Pull request by Midge-dev**
## Summary

_Product must have 80% or higher unit test coverage in each category: Lines, Functions, Statements, and Branches._
- https://depo-platform-documentation.scrollhelp.site/developer-docs/quality-assurance-standards
- This was addressed as a potential launch blocker for the Deactivated Pension form review page & _How do I apply?_ widget (see below)

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/66312
-  https://github.com/department-of-veterans-affairs/va.gov-team/issues/64136
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/63626

## Testing done

- Previous test coverage for the Pension app as a whole  was below QA standards (77%)
- Added tests and refactored existing tests to increase coverage (`vets-website/coverage/index.html`)

## Acceptance criteria
_Close the ticket when your Product Owner determines that you have sufficiently met the QA Standards.
If your team has additional questions or needs Platform help validating the issue, please comment on the ticket (#66312)_

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

File: src/applications/pensions/components/SectionField.jsx [hunk 1]
```diff
@@ -2,12 +2,18 @@ import React from 'react';
...
+  if (date) {
+    const [year, month, day] = date.split('-');
+    return `${month}/${day}/${year}`;
+  }
+  return date;
 };
 
-const formatPhoneNumber = num =>
-  `(${num.substr(0, 3)}) ${num.substr(3, 3)}-${num.substr(6)}`;
+export const formatPhoneNumber = num => {
```
Comment (DLarson-Oddball) [inline src/applications/pensions/components/SectionField.jsx:12]: Why would we want to return num in this context if it's falsy?

File: src/applications/pensions/components/SectionField.jsx [hunk 2]
```diff
@@ -2,12 +2,18 @@ import React from 'react';
 import { formatSSN } from 'platform/utilities/ui';
 
 export const convertDateFormat = date => {
-  const [year, month, day] = date.split('-');
-  return `${month}/${day}/${year}`;
+  if (date) {
```
Comment (DLarson-Oddball) [inline src/applications/pensions/components/SectionField.jsx:5]: Why would we want to return date in this context since it's falsy? Can we guard the invoking call instead?

File: src/applications/pensions/components/IntroductionPage.jsx [hunk 3]
```diff
@@ -9,10 +9,6 @@ class IntroductionPage extends React.Component {
     focusElement('.va-nav-breadcrumbs-list');
   }
 
-  goForward = () => {
```
Comment (DLarson-Oddball) [inline src/applications/pensions/components/IntroductionPage.jsx:12]: Why was this removed?

**Review by DLarson-Oddball — APPROVED** [review]
Just a few questions

File: src/applications/pensions/components/IntroductionPage.jsx (hunk 3 above)
Comment (Midge-dev) [reply src/applications/pensions/components/IntroductionPage.jsx:12]: goFoward wasn't being called anywhere in the component

File: src/applications/pensions/components/SectionField.jsx (hunk 1 above)
Comment (Midge-dev) [reply src/applications/pensions/components/SectionField.jsx:10]: I refactored :call_me_hand:

File: src/applications/pensions/components/SectionField.jsx (hunk 2 above)
Comment (Midge-dev) [reply src/applications/pensions/components/SectionField.jsx]: Same here I refactored

**Review by DLarson-Oddball — APPROVED** [review]
